tests/io/test_arrow.py

```diff
@@ -0,0 +1,117 @@
+import geopandas
+import dask_geopandas
+import dask.dataframe as dd
+
+import pytest
+from geopandas.testing import assert_geodataframe_equal, assert_geoseries_equal
+
+
+pa = pytest.importorskip("pyarrow")
+
+
+pytestmark = pytest.mark.filterwarnings(
+    "ignore:this is an initial implementation:UserWarning"
+)
+
+
+def test_read(tmp_path):
+    df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
+
+    # writing a partitioned dataset with geopandas (to not rely on roundtrip)
+    basedir = tmp_path / "dataset"
+    basedir.mkdir()
+    df.iloc[:100].to_feather(basedir / "data.0.feather")
+    df.iloc[100:].to_feather(basedir / "data.1.feather")
+
+    result = dask_geopandas.read_feather(basedir)
+    assert isinstance(result, dask_geopandas.GeoDataFrame)
+    assert result.npartitions == 2
+    assert result.crs == df.crs
+    assert result.spatial_partitions is not None
+    # TODO this reset_index should not be necessary
+    result_gpd = result.compute().reset_index(drop=True)
+    assert_geodataframe_equal(result_gpd, df)
+
+
+def test_write(tmp_path):
+    df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
+    ddf = dask_geopandas.from_geopandas(df, npartitions=4)
+
+    basedir = tmp_path / "dataset"
+    ddf.to_feather(basedir)
+
+    # each partition (4) is written as a feather file
+    paths = list(basedir.glob("*.feather"))
+    assert len(paths) == 4
+
+    # each individual file is a valid feather file
+    result_part0 = geopandas.read_feather(basedir / "part.0.feather")
+    result_part0.index.name = None
+    assert_geodataframe_equal(result_part0, df.iloc[:45])
+
+    # TODO geopandas doesn't actually support this for "feather" format
```

Comment:
Unclear if this commented block should be removed; since it is in a test case, might be better to move this to an issue and remove from here.

Reply:
Opened an issue -> https://github.com/geopandas/geopandas/issues/2348